crates/ruff_formatter/src/builders.rs

```diff
@@ -340,29 +337,25 @@ impl<Context> Format<Context> for SourcePosition {
     }
 }
 
-/// Creates a text from a dynamic string with its optional start-position in the source document.
+/// Creates a text from a dynamic string.
+///
 /// This is done by allocating a new string internally.
```

Comment:
It looks like this function doesn't allocate any more?

Reply:
This function doesn't, but it does when calling `Format`. I can improve the documentation.